packages/core/src/core/prompts.ts

```diff
@@ -486,132 +699,111 @@ I found the following 'app.config' files:
+</tool_call>
+(After reviewing existing tests and the file content)
+<tool_call>
+{"name": "${WriteFileTool.Name}", "arguments": {"path": "/path/to/someFile.test.ts"}}
+</tool_call>
+I've written the tests. Now I'll run the project's test command to verify them.
+<tool_call>
+{"name": "${ShellTool.Name}", "arguments": {"command": "npm run test"}}
+</tool_call>
+(After verification passes)
+All checks passed. This is a stable checkpoint.
+</example>
 
-    <current_plan>
-        <!-- The agent's step-by-step plan. Mark completed steps. -->
-        <!-- Example:
-         1. [DONE] Identify all files using the deprecated 'UserAPI'.
-         2. [IN PROGRESS] Refactor \`src/components/UserProfile.tsx\` to use the new 'ProfileAPI'.
-         3. [TODO] Refactor the remaining files.
-         4. [TODO] Update tests to reflect the API change.
-        -->
-    </current_plan>
-</state_snapshot>
+<example>
+user: Where are all the 'app.config' files in this project? I need to check their settings.
+model:
+<tool_call>
+{"name": "${GlobTool.Name}", "arguments": {"pattern": "./**/app.config"}}
+</tool_call>
+(Assuming GlobTool returns a list of paths like ['/path/to/moduleA/app.config', '/path/to/moduleB/app.config'])
+I found the following 'app.config' files:
+- /path/to/moduleA/app.config
+- /path/to/moduleB/app.config
+To help you check their settings, I can read their contents. Which one would you like to start with, or should I read all of them?
+</example>
 `.trim();
-}
-
-/**
- * Provides the system prompt for generating project summaries in markdown format.
- * This prompt instructs the model to create a structured markdown summary
- * that can be saved to a file for future reference.
- */
-export function getProjectSummaryPrompt(): string {
-  return `Please analyze the conversation history above and generate a comprehensive project summary in markdown format. Focus on extracting the most important context, decisions, and progress that would be valuable for future sessions. Generate the summary directly without using any tools.
-You are a specialized context summarizer that creates a comprehensive markdown summary from chat history for future reference. The markdown format is as follows:
-
-# Project Summary
-
-## Overall Goal
-<!-- A single, concise sentence describing the user's high-level objective -->
-
-## Key Knowledge
-<!-- Crucial facts, conventions, and constraints the agent must remember -->
-<!-- Include: technology choices, architecture decisions, user preferences, build commands, testing procedures -->
-
-## Recent Actions
-<!-- Summary of significant recent work and outcomes -->
-<!-- Include: accomplishments, discoveries, recent changes -->
-
-## Current Plan
-<!-- The current development roadmap and next steps -->
-<!-- Use status markers: [DONE], [IN PROGRESS], [TODO] -->
-<!-- Example: 1. [DONE] Set up WebSocket server -->
 
-`.trim();
+function getToolCallExamples(model?: string): string {
```

Comment:
Can we add an environment variable `QWEN_CODE_SYSTEM_PROMPT` with possible values: `general`, `qwen-coder`, or `qwen-vl`? If it’s not set, we can determine its value based on the model name.

For model name matching, we can use regex such as `qwen*-coder` or `qwen*-vl` to ensure compatibility with future open-source models such as `qwen3-vl-xxx`.

Reply:
sounds great!